sockeye/translate.py

```diff
@@ -62,6 +62,8 @@ def run_translate(args: argparse.Namespace):
                                         args.output,
                                         args.sure_align_threshold)
 
+    output_scores = True if args.output_type in C.OUTPUT_HANDLERS_SCORING else False
```

Comment:
`args.output_type in C.OUTPUT_HANDLERS_SCORING` ?

Reply:
What about just checking if the string "score" is in the output type? And then add a comment in `constants.py` noting that softmax is disabled for anything containing that keyword. i.e., 

    output_scores = C.OUTPUT_SCORE_KEYWORD in args.output_type?

Then in `constants.py`:

    # If this token is a substring of the output type, it disables the disabling of softmax in inference.py
    OUTPUT_SCORE_KEYWORD = "score"

Reply:
Checking for a string sounds a bit hacky. Keeping a separate list seems error prone for new output handlers that will be added in the future. How about making it explicit by extending the OutputHandler interface:

Something like:
```
class OutputHandler(ABC):
    """
    Abstract output handler interface
    """

    @abstractmethod
    def handle(self,
               t_input: inference.TranslatorInput,
               t_output: inference.TranslatorOutput,
               t_walltime: float = 0.):
        """
        :param t_input: Translator input.
        :param t_output: Translator output.
        :param t_walltime: Total wall-clock time for translation.
        """
        pass

    @abstractmethod
    def reports_score(self) -> bool:
        pass
```

Reply:
Yes this is much better.